### ISSUE TITLE
Directives should support selectors with dot notations

### ISSUE THREAD

**Issue by ThomasBurleson** [angular/angular#13351]
Currently only a single directive syntax is supported. Consider the Layout directive

```ts
@Directive({selector: '[fx-layout]'})
export class LayoutDirective {  
  @Input('fx-layout') layout = 'row';
  @Input('fx-layout.xs') layoutXs;
  @Input('fx-layout.gt-xs') layoutGtXs;
}
```

and the following markup usage:

```html
<div  fx-layout="row" fx-layout.xs="column"></div>
```

all ^ work fine... both properties `layout` and `layoutXs` are updated with values.

Now let's consider the use of multiple selectors with non-default used in markup

```ts
@Directive({selector: '[fx-layout], [fx-layout.xs], [fx-layout.gt-xs]'})
export class LayoutDirective {  
}
```

The following does NOT throw any errors:

```html
 <div [fx-layout]="direction" [fx-layout.xs]="'column'"></div>
```

## Issue

If consider this scenario:

```html
<div  [fx-layout.xs]="direction"></div>
````

or 

```html
<div  [fx-layout.xs]="'column'"></div>
````

In ^ this case, the expectation is a Layout directive instance would be created, and the `@Input('fx-layout.xs') layoutXs` would be assigned the value of the direction variable ( or 'column' string).

Instead, we get this error:

```sh
Template parse errors:(…) Error: Template parse errors:
Invalid property name 'fx-layout.xs' ("
          <div [ERROR ->][fx-layout.xs]="direction"></div>
```

> Note that the default value/input `@Input('fx-layout')` would not need to be defined in the markup, but would still be used (since its default value is assigned).

## Desired Result:

This should work without any errors:

```html
<div  [fx-layout.xs]="direction"></div>
````

and the directive instance property `layoutXs` will have is value updated accordingly.

> NOTE: this issue is **not** requesting that selectors with dot notations update @Input property maps. That would be a separate discussion. ;-).

**Comment by ThomasBurleson** [comment]
@vicb - as requested.

**Comment by vicb** [comment]
fixed by #13352
